make.sh

```diff
@@ -18,6 +18,8 @@
 
 set -e
 
+export LIGHTGBM_REPO_URL="${LIGHTGBM_REPO_URL:-https://github.com/microsoft/LightGBM}"
```

Comment:
Why can you just simply do this?
`export LIGHTGBM_REPO_URL="https://github.com/microsoft/LightGBM"`

Reply:
Because I want to let you override it :)

If you do LIGHTGBM_REPO_URL=https://sheng/lightgbm bash make.sh v2.3.1 it'll replace that.